src/plugins/intel_npu/src/al/include/intel_npu/al/icompiled_model.hpp

```diff
@@ -17,15 +18,17 @@ class ICompiledModel : public ov::ICompiledModel {
 public:
     using ov::ICompiledModel::ICompiledModel;
 
-    virtual const std::shared_ptr<const NetworkDescription>& get_network_description() const = 0;
+    virtual const NetworkDescription& get_network_description() const = 0;
 
     virtual const Config& get_config() const = 0;
 
     // Compiler is used for post-processing profiling data when using PERF_COUNT property
-    virtual const ov::SoPtr<ICompiler>& get_compiler() const = 0;
+    virtual const ICompiler& get_compiler() const = 0;
+
+    virtual IExecutor& get_executor() const = 0;
```

Comment:
Why we extend ICompiledModel model interface with executor?
This is a design change which is not clear from PR description by any means.

ICompiledModel is part of AL and is used in IMD backend. Long term maybe we can eliminate that dependency by re-implementing IMD-backend a a separate plugin, but until that point we should not extend the interface beyond the common functionality.

Reply:
ICompiledModel is also part of the zero backend as well. Since the executor is part of the compiled model it just doesn't make sense to share it as a function parameter together with the ICompiledModel as well as with the backend methods.

Reply:
So far executor is a member of the CompiledModel implementation but not exposed in ICompiledModel interface.
This looks like convenience of implementation is a motivation to make executor go thru more layers of abstraction before the downcast, not sure that's a good tradeoff in terms of code readability. We increase executor scope even though we don't have the concrete type.